geogfn: use Contains instead of IntersectsCell for point intersects

otan opened the pull request:
Release justification: low risk performance enhancement

Resolves #69761

Release note (performance improvement, sql change): geography point
intersection is slightly faster as we use a speedier intersection
function. However, this also means point intersection is "stricter",
requiring points be equal rather than having a small possible leeway
(<1cm) to be considered intersecting.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69762)
<!-- Reviewable:end -->

otan commented:
there is a corollary to do this -- we are stricter in intersects in that we no longer allow a small leeway (<1cm) to be considered intersecting.

otan committed:
geogfn: use Contains instead of IntersectsCell for point intersects

Release justification: low risk performance enhancement

Release note (performance improvement, sql change): geography point
intersection is slightly faster as we use a speedier intersection
function. However, this also means point intersection is "stricter",
requiring points be equal rather than having a small possible leeway
(<1cm) to be considered intersecting.

sumeerbhola reviewed (commented):
Do we have any benchmarking numbers that justify this change in semantics?
And is the concern here that two points that are infinitesimally close to each other but straddle a cell boundary will be discretized to adjacent cells at level 30 and will no longer be considered intersecting? Are all adjacent cells at level 30 considered intersecting prior to this change?
Setting aside backports, I suppose this could be an unpleasant surprise to an existing user when they upgrade.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69762#-:-Mj2NTdS0zmd2ts8Qvnq:b-ml0b3i)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
i don't have  a compelling reason to merge this atm and don't have too much time to experiment. i'll close it for now, but leave an issue
